hoomd/mpcd/CollisionMethod.cu

```diff
@@ -0,0 +1,263 @@
+// Copyright (c) 2009-2025 The Regents of the University of Michigan.
+// Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+/*!
+ * \file mpcd/CollisionMethod.cu
+ * \brief Defines GPU functions and kernels used by mpcd::CollisionMethod
+ */
+
+#include "CollisionMethod.cuh"
+#include "hoomd/ParticleData.cuh"
+
+namespace hoomd
+    {
+namespace mpcd
+    {
+namespace gpu
+    {
+namespace kernel
+    {
+
+__global__ void store_initial_embedded_group_velocities(Scalar4* d_initial_vel,
+                                                        const Scalar4* d_velocity,
+                                                        const unsigned int* d_embed_group,
+                                                        const unsigned int num_group)
+    {
+    // one thread per particle
+    unsigned int idx = blockIdx.x * blockDim.x + threadIdx.x;
```

Comment:
This can be const, although I may have forgotten to do so in older code
```suggestion
    const unsigned int idx = blockIdx.x * blockDim.x + threadIdx.x;
```